pandas/core/util/hashing.py

```diff
@@ -164,6 +164,29 @@ def hash_tuples(vals, encoding='utf8', hash_key=None):
     return h
 
 
+def hash_tuple(val, encoding='utf8', hash_key=None):
+    """
+    Hash a single tuple efficiently
+
+    Parameters
+    ----------
+    val : single tuple
+    encoding : string, default 'utf8'
+    hash_key : string key to encode, default to _default_hash_key
+
+    Returns
+    -------
+    hash
+
+    """
+    hashes = (hash_array(np.array([v]), encoding=encoding, hash_key=hash_key,
```

Comment:
for a *single* element ``np.array`` will work, but if you have multiple elements it will not, use this insteadIn [10]: from pandas.core.dtypes.cast import infer_dtype_from_array

```
from pandas.core.dtypes.cast import infer_dtype_from_array

 [12]: infer_dtype_from_array(['foo', np.nan])
Out[12]: (numpy.object_, ['foo', nan])

In [14]: dtype, arr = infer_dtype_from_array(['foo', np.nan])

In [15]: np.array(arr, dtype=dtype)
Out[15]: array(['foo', nan], dtype=object)

In [16]: np.array(['foo', np.nan])
Out[16]: 
array(['foo', 'nan'], 
      dtype='<U3')

```